src/cassy.py

```diff
@@ -557,3 +557,553 @@ def change_user_email(username, new_email):
     except Exception as e:
         raise Exception('Transaction Error Occurred: '.format(str(e)))
 
+
+
+def verify_authenticated_admin(username, auth_token):
+    """
+    Verifies if supplied username is an admin and is authenticated.
+    """
+
+    session.row_factory = named_tuple_factory
+    table = str(os.environ.get('DB_USER_TABLE'))
+    parameters = {
+        'username': username,
+    }
+    query = (
+        'SELECT admin FROM {} WHERE username=?;'
+    )
+    prepared_statement = session.prepare(
+        query.format(table)
+    )
+
+    try:
+        verified = str(verify_auth_token(auth_token))
+        if (verified != username):
+            return False
+        rows = session.execute(
+            prepared_statement,
+            parameters
+        )
+        if not rows:
+            return False
+        return True
+    # Known exception
+    except PlantalyticsAuthException as e:
+            return False
+    except PlantalyticsException as e:
+        raise e
+    # Unknown exception
+    except Exception as e:
+        raise Exception('Transaction Error Occurred: '.format(str(e)))
+
+
+def get_user_info(username):
+    """
+    Obtains email, user id, and vineyard ids for submitted user.
+    """
+
+    session.row_factory = named_tuple_factory
+    table = str(os.environ.get('DB_USER_TABLE'))
+    parameters = {
+        'username': username,
+    }
+    query = (
+        'SELECT email, userid, vineyards FROM {} WHERE username=?;'
+    )
+    prepared_statement = session.prepare(
+        query.format(table)
+    )
+
+    try:
+        rows = session.execute(
+            prepared_statement,
+            parameters
+        )
+        if not rows:
+            raise PlantalyticsAuthException(USER_INVALID)
+        user_info = {
+            'email': rows[0].email,
+            'userid': rows[0].userid,
+            'vineyards': rows[0].vineyards,
+        }
+        return user_info
+    # Known exception
+    except PlantalyticsException as e:
+        raise e
+    # Unknown exception
+    except Exception as e:
+        raise Exception('Transaction Error Occurred: '.format(str(e)))
+
+
+def create_new_user(new_user_info):
+    """
+    Creates new user in DB using the submitted info.
+    """
+
+    session.row_factory = named_tuple_factory
+    table = str(os.environ.get('DB_USER_TABLE'))
+    parameters = {
+            'username': new_user_info.get('username', ''),
+            'password': new_user_info.get('password', ''),
+            'email': new_user_info.get('email', ''),
+            'admin': new_user_info.get('admin', ''),
+            'enable': new_user_info.get('enable', ''),
+            'securitytoken': new_user_info.get('securitytoken', ''),
+            'subenddate': new_user_info.get('subenddate', ''),
+            'userid': int(new_user_info.get('userid', '')),
+            'vineyards': new_user_info.get('vineyards', ''),
```

Comment:
Is there a mechanism that does some kind of checking about these parameters? Like that userids are positive integers w/o duplicate (or whatever), sub end date is actually a date in the future, etc.